### DISCUSSION TITLE
Can anyone access to source code?

### DISCUSSION THREAD

**Discussion by steam3d** [WeblateOrg/weblate#5184] [Q&A] [answered]
For case when you use private repository and added this repository to weblate.

**Comment by nijel** [comment, marked as answer]
Only users with appropriate permission can access the repository exported by Weblate, see https://docs.weblate.org/en/latest/admin/optionals.html#git-exporter and https://docs.weblate.org/en/latest/admin/access.html#managing-per-project-access-control

**Reply by steam3d** [reply]
If you add a private repo to public project weblate, everyone can clone it. Does it right behaviour? isn't it just the translation text string files that need to be available?

**Reply by nijel** [reply]
Only users who have permission for that can access it. In the default setup the Users group (default one for created users) or anonymous account (user for unauthenticated users) have that permission, but you are free to change that to match your expectations.

**Reply by steam3d** [reply]
I think you should mention this in weblate. I accidentally noticed when I saw the link to the internal weblate repository. This is really not obvious behavior.
![Group 436](https://user-images.githubusercontent.com/30986874/105192111-d7217880-5b48-11eb-8682-0c3a0124bc23.png)

**Reply by nijel** [reply]
I've added such explanation in ad79c418aacc5f7ea88450a48535d0b2fa8c058d.
